pkg/security/ptracer/cws.go

```diff
@@ -492,25 +553,44 @@ func StartCWSPtracer(args []string, probeAddr string, creds Creds, verbose bool)
 			msg.SeqNum = seq
 			msg.NSID = nsid
 
-			logDebugf("sending message: %s", msg)
-
 			if probeAddr != "" {
-				data, err := msgpack.Marshal(msg)
-				if err != nil {
-					logErrorf("unable to marshal message: %v", err)
-					return
-				}
-
-				// write size
-				var size [4]byte
-				native.Endian.PutUint32(size[:], uint32(len(data)))
-				if _, err = client.Write(size[:]); err != nil {
-					logErrorf("unabled to send size: %v", err)
-				}
-
-				if _, err = client.Write(data); err != nil {
-					logErrorf("unabled to send message: %v", err)
+				if clientReady {
```

Comment:
what about just using the msgChan with a bigger size ? Here I have the impression that you can have out of order messages between open and exec events. We could put the clientReady in the "initGRPCcon" function (need to add a mutex)